File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
         Stack<Character> lifo= new Stack<>();
        for (Character c : s.toCharArray()) {
            if (c=='('||c=='['||c=='{'){
                lifo.push(c);
            }else {
                if (lifo.isEmpty()) return false;
                Character pop = lifo.pop();
                if (c==')'&&pop!='(') return false;
                if (c==']'&&pop!='[') return false;
                if (c=='}'&&pop!='{') return false;
            }
        }
        return lifo.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        mp = {'(':')', '{':'}','[':']'}
        stk = []
        for i in s:
            if i in mp:
                stk.append(i)
            elif len(stk) == 0 or mp[stk.pop()] != i:
                return False
        return len(stk) == 0
                    
        